api/src/main/java/com/flipkart/poseidon/api/APILegoSet.java

```diff
@@ -65,10 +65,7 @@ public Buildable getBuildable(Request request) throws LegoSetException, ElementN
         String completeUrl = ApiHelper.getUrlWithHttpMethod(poseidonRequest.getUrl(), httpMethod);
         Buildable buildable = trie.get(getKeysForTrie(completeUrl));
         if (buildable == null) {
-            buildable = trie.get(getKeysForTrie(poseidonRequest.getUrl()));
-            if (buildable == null) {
-                throw new ElementNotFoundException("Buildable not found for given url: " + poseidonRequest.getUrl());
-            }
+            throw new ElementNotFoundException("Buildable not found for given url: " + poseidonRequest.getUrl());
```

Comment:
With this change, httpMethod is now mandatory in API definition. Please update https://github.com/flipkart-incubator/Poseidon/wiki/Migration-Guide-%5B3.x-to-4.x%5D